### ISSUE TITLE
`f1` seems wrong

### ISSUE THREAD

**Issue by OverLordGoldDragon** [PyTorchLightning/metrics#629]
```python
from torchmetrics.functional import f1
from sklearn.metrics import f1_score

target = torch.tensor([1, 1, 0, 0])
preds  = torch.tensor([1, 0, 0, 0])
print(f1(preds, target))
print(f1_score(target.numpy(), preds.numpy())
```
```python
tensor(0.7500)
0.6666666666666666
```
Unsure of intermediate processing steps but the following, interpreted directly, is incorrect:

https://github.com/PyTorchLightning/metrics/blob/2c8e46f87cb67186bff2c7b94bf1ec37486873d4/torchmetrics/functional/classification/stat_scores.py#L64-L67

TP is defined as "label is 1 and prediction is 1", which is, for binary classification,

```python
tp = ((target == 1) * (preds == 1)).sum()
```
Bug, or am I misusing the method?

**Comment by stancld** [comment]
The difference is in the default `average` method. For `torchmetrics`, there is `average="micro"`, whereas `sklearn` has `average="binary"`.  The latter method is not directly supported by TM, however, according to the `sklearn` [docs](https://github.com/scikit-learn/scikit-learn/blob/0d378913be6d7e485b792ea36e9268be31ed52d0/sklearn/metrics/_classification.py#L1033)
```
 average : {'micro', 'macro', 'samples','weighted', 'binary'} or None, \
            default='binary'
        This parameter is required for multiclass/multilabel targets.
        If ``None``, the scores for each class are returned. Otherwise, this
        determines the type of averaging performed on the data:
        ``'binary'``:
            Only report results for the class specified by ``pos_label``.
            This is applicable only if targets (``y_{true,pred}``) are binary.
``` 

it is nothing else than
```
>>> f1(target, preds, average="none", num_classes=2)[1]
tensor(0.6667)
```
